mlflow/pyfunc/model.py

```diff
@@ -226,13 +226,29 @@ class ChatModel(PythonModel, metaclass=ABCMeta):
 
     See the documentation of the ``predict()`` method below for details on that parameters and
     outputs that are expected by the ``ChatModel`` API.
+
+    Warning: Before or during the MLflow 2.20 release, we will be requiring a predict_stream
```

Comment:
Let's change this to a `.. warning::` block. 
Also, as @harupy said, we can just remove the version mention since it's a little vague.

Can we also update the `predict_stream` warning to talk about the return type changing (the API exists right now, it's just "fake" in that it's returning a generator on the synchronous response - this will be changing to returning an actual generator of Chunk responses as a 'real' streaming API should